webapp/components/Map/MapBaseLayerPeriodSelector.js

```diff
@@ -32,14 +40,62 @@ export const MapBaseLayerPeriodSelector = () => {
 
   const i18n = useI18n()
   const user = useUser()
-  const [state, setState] = useState({ ready: false, periods: [], periodByValue: {}, selectedPeriodValue: null })
+  const [state, setState] = useState({
+    ready: false,
+    periods: [],
+    periodByValue: {},
+    selectedPeriodValueLeft: null,
+    selectedPeriodValueRight: null,
+    leftChecked: false,
+    rightChecked: false,
+    procLeft: Proc.rgb,
+    procRight: Proc.rgb,
+  })
 
   const apiKey = User.getMapApiKey({ provider })(user)
 
-  const { ready, periods, periodByValue, selectedPeriodValue } = state
+  const {
+    ready,
+    periods,
+    periodByValue,
+    selectedPeriodValueLeft,
+    selectedPeriodValueRight,
+    leftChecked,
+    rightChecked,
+    procLeft,
+    procRight,
+  } = state
+
+  const getLayer = (isLeft) => {
+    if (isLeft) {
+      return Object.values(map._layers).find((layer) => layer.options.id !== 'right')
+    } else {
+      return Object.values(map._layers).find((layer) => layer.options.id === 'right')
+    }
+  }
+  const initSideBySide = () => {
+    if (User.isSystemAdmin(user)) {
```

Comment:
you can reduce the complexity and the nested levels if you type:

if(!User.isSystemAdmin(user)) return
...

Reply:
Oh, true. Just put it there quickly since it will be removed for the actual publish anyway.